src/NewUser.js

```diff
@@ -14,45 +14,42 @@ class NewUser extends Component {
 
 
   handleChange(event) {
-    console.log(event.target.value)
     this.setState({input: event.target.value});
   }
 
-  handleSubmit(event) {
-    cookie.save("username", this.state.input);
-    this.setState({showModal: false})
+  handleSubmit() {
+    this.props.onSubmit(this.state.input)
   }
 
 
   render() {
-    const {input, showModal} = this.props
-    console.log(showModal)
+    const {showModal} = this.props
+    const {input} = this.state
+    const nameTyped = input === ""
```

Comment:
This is setting `nameTyped` to be a boolean. If input is an emtpy string `""` then boolean will disable the button. The default for input is an empty string `""`, which is set in the constructor above.  
```
<Button disabled={nameTyped} />
```